### PR TITLE
Bug 1774616: Remove insecureSkipTLSVerifyBackend bypass for BuildLogOptions

### PR DESCRIPTION
**Pull request by coreydaley**
Remove the bypass for InsecureSkipTLSVerifyBackend when comparing
BuildLogOptions and PodLogOptions
Depends on https://github.com/openshift/api/pull/652

### CODE REVIEW

**Comment by coreydaley** [comment]
/hold
Waiting on https://github.com/openshift/api/pull/652

File: pkg/build/apis/build/log_options_test.go [hunk 1]
```diff
@@ -14,10 +14,6 @@ func TestLogOptionsDrift(t *testing.T) {
 	for i := 0; i < popts.NumField(); i++ {
 		// Verify name
 		name := popts.Field(i).Name
-		// TODO: Add this option to deployment config log options
```
Comment (bparees) [inline pkg/build/apis/build/log_options_test.go:17]: i hope this comment was wrong and should have said buildconfig log options.

Comment (coreydaley) [reply pkg/build/apis/build/log_options_test.go:17]: Looking at the rest of the file it is comparing BuildLogOptions and PodLogOptions, so I would hope so ;)

**Comment by bparees** [comment]
@coreydaley is there no work that needs to be done to actually wire the build log request logic up to use the new field?

**Comment by coreydaley** [comment]
> @coreydaley is there no work that needs to be done to actually wire the build log request logic up to use the new field?

Quite possibly, working on the rest of this pr now.

**Comment by coreydaley** [comment]
@bparees ptal

**Comment by bparees** [comment]
so the field is not used?  oc logs has a --insecure field, i'd expect that when that field is set and you're doing an "oc logs build/foo" invocation, the insecure field gets propagated into this new field?

**Comment by bparees** [comment]
/hold

```
func BuildToPodLogOptions(opts *buildapi.BuildLogOptions) *kapi.PodLogOptions {
```

needs to be updated to map this field, i believe.

**Comment by coreydaley** [comment]
@bparees the field has been updated

**Comment by bparees** [comment]
as @adambkaplan noted on https://github.com/openshift/oc/pull/433, the vendor changes should be in a bump commit.

**Comment by coreydaley** [comment]
@bparees fixed commit message

**Pull request merged**